File: JULY-2025/09-07-2025/p3439.java
import java.util.*;
class Solution {
    public int maxFreeTime(int eventTime, int k, int[] startTime, int[] endTime) {
        ArrayList<Integer> list = new ArrayList<>();
         list.add(startTime[0]);
        for(int i=1;i<startTime.length;i++){
           list.add(startTime[i]-endTime[i-1]);
        }
        list.add(eventTime - endTime[endTime.length-1]);
        int i=0;
        int j=0;
        int maxSum = 0;
        int currSum = 0;
        int n = list.size();
        while(j<n){
            currSum+= list.get(j);

            if(i<n && (j-i+1) > k+1){
                currSum -= list.get(i);
                i++;
            }

            maxSum = Math.max(maxSum, currSum);
            j++;
        }
        return maxSum;
    }
}

File: JULY-2025/09-07-2025/p3439.ts
function maxFreeTime(eventTime: number, k: number, startTime: number[], endTime: number[]): number {
    let scheduledTime = 0;
    startTime.unshift(0);
    endTime.unshift(0);
    startTime.push(eventTime);
    endTime.push(eventTime);
    for(let i = 0; i < k; i++) {
        scheduledTime += endTime[i] - startTime[i];
    }
    let maxTime = startTime[k] - scheduledTime;
    for(let i = 1; i < startTime.length - k; i++) {
        scheduledTime -= endTime[i - 1] - startTime[i - 1];
        scheduledTime += endTime[i + k - 1] - startTime[i + k - 1];
        maxTime = Math.max(maxTime, startTime[i + k] - endTime[i - 1] - scheduledTime);
    }
    return maxTime;
};